bugsnag-android-core/src/main/java/com/bugsnag/android/DeviceData.java

```diff
@@ -253,25 +254,14 @@ private String retrieveUniqueInstallId() {
     }
 
     /**
-     * Get the free disk space on the smallest disk
+     * Get the usable disk space on internal storage's data directory
      */
-    @Nullable
-    @SuppressWarnings("deprecation") // ignore blockSizeLong suggestions for now (requires API 18)
-    private Long calculateFreeDisk() {
-        try {
-            StatFs externalStat = new StatFs(Environment.getExternalStorageDirectory().getPath());
-            long externalBytesAvailable =
-                (long) externalStat.getBlockSize() * (long) externalStat.getBlockCount();
-
-            StatFs internalStat = new StatFs(Environment.getDataDirectory().getPath());
-            long internalBytesAvailable =
-                (long) internalStat.getBlockSize() * (long) internalStat.getBlockCount();
-
-            return Math.min(internalBytesAvailable, externalBytesAvailable);
-        } catch (Exception exception) {
-            Logger.warn("Could not get freeDisk");
-        }
-        return null;
+    @SuppressLint("UsableSpace")
+    private long calculateFreeDisk() {
+        // for this specific case we want the currently usable space, not
+        // StorageManager#allocatableBytes() as the UsableSpace lint inspection suggests
+        File dataDirectory = Environment.getDataDirectory();
```

Comment:
Also why the move from statfs?

Reply:
`getFilesDir()` would not include the cache directory and so would not provide a true value for the amount of free disk in internal storage.

The statfs calculation was wrong and would return the total amount of usable disk - it therefore felt preferable to rely on existing Android Framework APIs rather than implementing our own solution.

Reply:
How would not including the cache directory affect the free disk on the volume?

The statfs calculation could be updated to call `getFreeBlocksLong` and it would work.

Reply:
I have good news! `File.getUsableSpace()` for any directory in internal memory is the same as `StatFs.getAvailableBlocks() / StatFs.getBlockCount()`.

<details>
<summary>Source code snippet</summary>

```java
   // File.java
   /**
     * Returns the number of usable free bytes on the partition containing this path.
     * Returns 0 if this path does not exist.
     *
     * <p>Note that this is likely to be an optimistic over-estimate and should not
     * be taken as a guarantee your application can actually write this many bytes.
     * On Android (and other Unix-based systems), this method returns the number of free bytes
     * available to non-root users, regardless of whether you're actually running as root,
     * and regardless of any quota or other restrictions that might apply to the user.
     * (The {@code getFreeSpace} method returns the number of bytes potentially available to root.)
     *
     * @since 1.6
     */
    public long getUsableSpace() {
        try {
            StructStatFs sb = Libcore.os.statfs(path);
            return sb.f_bavail * sb.f_bsize; // non-root free block count * block size in bytes.
        } catch (ErrnoException errnoException) {
            return 0;
        }
    }

    // StatFs.java
    /**
     * The number of blocks that are free on the file system and available to
     * applications. This corresponds to the Unix {@code statvfs.f_bavail} field.
     */
    public long getAvailableBlocksLong() {
        return mStat.f_bavail;
    }
```

</details>

I understand now after reading through the `StatFs` docs that the previous implementation computed total disk space instead of free disk space, which not obvious from the PR description for why this change is happening. As the vast majority of the time the app is installed in internal memory, removing the external space calculation also seems reasonable (I strongly suspect that's why its in there). The only change I'd require from this PR is a better changelog description. This is a fix which displays the correct value for free disk space in reports.